packages/hub/routes/email-card-drop-verify.ts

```diff
@@ -0,0 +1,67 @@
+import Koa from 'koa';
+import autoBind from 'auto-bind';
+import { query } from '../queries';
+import { inject } from '@cardstack/di';
+
+export default class EmailCardDropVerifyRoute {
+  emailCardDropRequestQueries = query('email-card-drop-requests', { as: 'emailCardDropRequestQueries' });
+  emailCardDropRequestSerializer = inject('email-card-drop-request-serializer', {
+    as: 'emailCardDropRequestSerializer',
+  });
+
+  workerClient = inject('worker-client', { as: 'workerClient' });
+
+  constructor() {
+    autoBind(this);
+  }
+
+  async get(ctx: Koa.Context) {
+    if (!ctx.request.query.eoa) {
+      ctx.status = 400;
+      ctx.body = 'eoa is required';
+      return;
+    }
+
+    let ownerAddress = ctx.request.query.eoa as string;
+    let verificationCode = (ctx.request.query['verification-code'] as string) || '';
+
+    // Optimistically mark the request as claimed to prevent stampede attack
+    let updatedRequest = await this.emailCardDropRequestQueries.claim({ ownerAddress, verificationCode });
```

Comment:
Do we set the `claimed_at` to `null` if the `drop-card` job fails?

Reply:
yes, I’ll make note of that in the `drop-card` issue, thanks.